frame/utility/src/lib.rs

```diff
@@ -15,48 +15,620 @@
+		/// transaction index) of the first approval transaction.
+		/// - `call`: The call to be executed.
+		///
+		/// NOTE: Unless this is the final approval, you will generally want to use
+		/// `approve_as_multi` instead, since it only requires a hash of the call.
+		///
+		/// Result is equivalent to the dispatched result if it is the final `Ok` approval. If the
+		/// dispatch fails, them the operation will still be open and may be retried by issuing
+		/// another approval. `cancel_as_multi` may be used to cancel the operation and return the
+		/// deposit which may be useful in the case of a call that can never succeed.
+		///
+		/// # <weight>
+		/// - `O(S + Z + Call)`.
+		/// - Up to one balance-reserve or unreserve operation.
+		/// - One passthrough operation, one insert, both `O(S)` where `S` is the number of
+		///   signatories. `S` is capped by `MaxSignatories`, with weight being proportional.
+		/// - One call encode & hash, both of complexity `O(Z)` where `Z` is tx-len.
+		/// - One encode & hash, both of complexity `O(S)`.
+		/// - Up to one binary search and insert (`O(logS + S)`).
+		/// - I/O: 1 read `O(S)`, up to 1 mutate `O(S)`. Up to one remove.
+		/// - One event.
+		/// - The weight of the `call`.
+		/// - Storage: inserts one item, value size bounded by `MaxSignatories`, with a
+		///   deposit taken for its lifetime of
+		///   `MultisigDepositBase + threshold * MultisigDepositFactor`.
+		/// # </weight>
+		#[weight = <MultiPassthrough<<T as Trait>::Call, T::AccountId, Option<Timepoint<T::BlockNumber>>>>::new()]
+		fn as_multi(origin,
+			threshold: u16,
+			other_signatories: Vec<T::AccountId>,
+			maybe_timepoint: Option<Timepoint<T::BlockNumber>>,
+			call: Box<<T as Trait>::Call>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(threshold >= 1, Error::<T>::ZeroThreshold);
+			let max_sigs = T::MaxSignatories::get() as usize;
+			ensure!(!other_signatories.is_empty(), Error::<T>::TooFewSignatories);
+			ensure!(other_signatories.len() < max_sigs, Error::<T>::TooManySignatories);
+			let signatories = Self::ensure_sorted_and_insert(other_signatories, who.clone())?;
+
+			let id = Self::multi_account_id(&signatories, threshold);
+			let call_hash = call.using_encoded(blake2_256);
+
+			if let Some(mut m) = <Multisigs<T>>::get(&id, call_hash) {
+				let timepoint = maybe_timepoint.ok_or(Error::<T>::NoTimepoint)?;
+				ensure!(m.when == timepoint, Error::<T>::WrongTimepoint);
+				if let Err(pos) = m.approvals.binary_search(&who) {
+					if (m.approvals.len() as u16) + 1 < threshold {
+						m.approvals.insert(pos, who.clone());
+						<Multisigs<T>>::insert(&id, call_hash, m);
+						Self::deposit_event(RawEvent::MultisigApproval(who, timepoint, id));
+						return Ok(())
+					}
+				} else {
+					if (m.approvals.len() as u16) < threshold {
+						Err(Error::<T>::AlreadyApproved)?
+					}
+				}
+				call.dispatch(frame_system::RawOrigin::Signed(id.clone()).into())?;
```

Comment:
Due to `?` it is possible to execute the call multiple times if it returns `Err`. This might be not desirable since it is not guaranteed nor required for a call to be side-effect free if it returns `Err`.

Reply:
it is not guaranteed, but it is good practice.